lib/schema-fs-loader.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const path = require('path');
+const vm = require('vm');
+const metasync = require('metasync');
+const fs = require('fs');
+
+const { create } = require('./schema');
+const { all: decorators } = require('./decorators');
+
+const SCRIPT_TIMEOUT = 2000;
+const DOMAINS_SCHEMA = 'domains.schema';
+
+const moduleDir = path.dirname(__dirname);
+const schemasDir = path.join(moduleDir, 'schemas');
+
+const processSchemaFile = (filename, decorators, source, api = {}) => {
+  const sandbox = Object.assign({}, decorators, { api });
+  sandbox.global = sandbox;
+  const code = `'use strict';\n(${source})`;
+  const options = {
+    filename,
+    timeout: SCRIPT_TIMEOUT,
+    lineOffset: 1,
+  };
+  return vm.runInNewContext(code, sandbox, options);
+};
+
+// Load schema file from the filesystem
+//   filepath - <string> full path to the schema file
+//   api - <Object> object to be passed as 'api' during loading
+//   callback - <Function> to report the result
+//     error - <Error> error or null
+//     name - <string> schema name
+//     schema - <Object> parsed schema
+const loadSchema = (filepath, api, callback) => {
+  fs.readFile(filepath, (err, source) => {
+    if (err) {
+      callback(err);
+      return;
+    }
+    const name = path.basename(filepath, '.schema');
+    let exports;
+    try {
+      exports = processSchemaFile(filepath, decorators, source, api);
+    } catch (e) {
+      callback(e);
+      return;
+    }
+    callback(null, name, exports);
+  });
+};
+
+const loadFiles = (files, api, callback) => {
+  const schemas = [];
+  const domainsPath = path.join(schemasDir, 'metaschema', DOMAINS_SCHEMA);
```

Comment:
Should `'metaschema'` be a named constant too?

Reply:
I extracted whole domainsPath and inlined the variables, wdyt?